### ISSUE TITLE
Packaging a Box with Include File that has Symlinks

### ISSUE THREAD

**Issue by lloy0076** [hashicorp/vagrant#3200]
TL;DR

It appears that the "vagrant package" subcommand fails on my OS X Mavericks machine when the include file directory contains symlinks. I do not know whether this is by design or by error.
1. If it's by error, it's a bug.

My apologies, but I don't know quite enough how to fix it.
1. If it is an error is there anyway to avoid the "not helpful to non-Ruby programmers" stack trace and preface it with something useful.

LONG

I keep getting:

```
[Arkade Centos 64 Take II] Packaging additional file: ./DEPLOY-SCRIPTS-final/
It appears that the file destination, /Users/lloy0076/.vagrant.d/tmp/1394758016/include/DEPLOY-SCRIPTS-final/manifests, is some type of @##$@# symlink at path ./DEPLOY-SCRIPTS-final/manifests.
/Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1354:in `symlink': File exists - (misc/manifests/, /Users/lloy0076/.vagrant.d/tmp/1394758016/include/DEPLOY-SCRIPTS-final/manifests) (Errno::EEXIST)
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1354:in `copy'
```

The line with the masked out expletive is my own debugging as I'm not a fluent Ruby programmer.

It appears that Vagrant does _not_ like to include symlinks when one does

```
vagrant package --base some_base --out some_file --include oh_no_a_dir_with_symlinks
```

So this directory might contain something like this:

```
dir-
  misc/real_file
  link_to_file [ls -l would show this as a link to misc/real_file]
```

Vagrant will percolate the Ruby exception up to the user.

I honestly don't know what Vagrant is attempting to do at this time but:
1. If Vagrant really can't handle symlinks in the include files, perhaps the error could be a little nicer.
2. If it really should (it probably shouldn't given that a symlink could link to anything at all), then it's a bug.

I suspect a full trace would be used:

```
[Arkade Centos 64 Take II] Packaging additional file: ./DEPLOY-SCRIPTS-final/
It appears that the file destination, /Users/lloy0076/.vagrant.d/tmp/1394758016/include/DEPLOY-SCRIPTS-final/manifests, is some type of !@ symlink at path ./DEPLOY-SCRIPTS-final/manifests.
/Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1354:in `symlink': File exists - (misc/manifests/, /Users/lloy0076/.vagrant.d/tmp/1394758016/include/DEPLOY-SCRIPTS-final/manifests) (Errno::EEXIST)
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1354:in `copy'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:463:in `block in copy_entry'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1480:in `call'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1480:in `wrap_traverse'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1483:in `block in wrap_traverse'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1482:in `each'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1482:in `wrap_traverse'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:460:in `copy_entry'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:435:in `block in cp_r'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1553:in `block in fu_each_src_dest'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1562:in `block in fu_each_src_dest0'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1560:in `each'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1560:in `fu_each_src_dest0'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:1551:in `fu_each_src_dest'
    from /Applications/Vagrant/embedded/lib/ruby/2.0.0/fileutils.rb:434:in `cp_r'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/general/package.rb:70:in `block in copy_include_files'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/general/package.rb:61:in `each'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/general/package.rb:61:in `copy_include_files'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/general/package.rb:82:in `compress'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/general/package.rb:39:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/package.rb:23:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/builtin/synced_folder_cleanup.rb:28:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/prepare_nfs_valid_ids.rb:12:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/clear_forwarded_ports.rb:13:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/builtin/call.rb:57:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/setup_package_files.rb:46:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/builder.rb:116:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/runner.rb:69:in `block in run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/util/busy.rb:19:in `busy'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/runner.rb:69:in `run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/builtin/call.rb:51:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/warden.rb:34:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/builder.rb:116:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/runner.rb:69:in `block in run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/util/busy.rb:19:in `busy'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/action/runner.rb:69:in `run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/machine.rb:147:in `action'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/commands/package/command.rb:83:in `package_vm'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/commands/package/command.rb:66:in `package_base'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/plugins/commands/package/command.rb:42:in `execute'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/cli.rb:38:in `execute'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/lib/vagrant/environment.rb:484:in `cli'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.4.3/bin/vagrant:127:in `<top (required)>'
    from /Applications/Vagrant/bin/../embedded/gems/bin/vagrant:23:in `load'
    from /Applications/Vagrant/bin/../embedded/gems/bin/vagrant:23:in `<main>'
Isengaard-2:with_files lloy0076$ cd DEPLOY-SCRIPTS-final/
```
